src/printer.js

```diff
@@ -1,4 +1,30 @@
 // https://prettier.io/docs/en/plugins.html#printers
-const genericPrint = (path, options) => options.originalText;
+const { concat, hardline, join } = require('prettier').doc.builders;
+
+const space = ' ';
+
+const genericPrint = (path, options, print) => {
+  const node = path.getValue();
+
+  switch (node.type) {
+    case 'SourceUnit':
+      return join(hardline, path.map(print, 'children'));
```

Comment:
as it is `children` containing `PragmaDirective` and other node types; so if I understood properly, the `map` function invokes the print function against every element in the children collection?

Reply:
That’s correct! One pitfall to be aware of is that there’s only one `path` object that gets mutated when you call `path.map`/`path.call` and passed to `print`.